cblite-tests/e2e/replicator-test.ts

```diff
@@ -310,39 +381,231 @@ export class ReplicatorTests extends TestCase {
    *
    * @returns {Promise<ITestResult>} A promise that resolves to an ITestResult object which contains the result of the verification.
    */
-  async testSingleShotPushFilter(): Promise<ITestResult> {
+async testSingleShotPushFilter(): Promise<ITestResult> {
+  try {
+    // Create test documents with unique IDs
+    const doc1Id = `test-doc-1-${Date.now()}`
+    const doc1 = this.createDocumentWithIdAndData(doc1Id, {
+      name: 'should-push', 
+      documentType: 'project', 
+      team: 'team1'
+    })
+    await this.defaultCollection.save(doc1)
+
+    const doc2Id = `test-doc-2-${Date.now()}`
+    const doc2 = this.createDocumentWithIdAndData(doc2Id, {
+      name: 'not-push', 
+      documentType: 'project', 
+      team: 'team1'
+    })
+    await this.defaultCollection.save(doc2)
+
+    // Create collection config with push filter
+    const collectionConfig = new CollectionConfig();
+    collectionConfig.setPushFilter((doc) => {
+      "show source";
```

Comment:
Can you explain what is the "show source" here for?

Reply:
It was needed to tell RN to not change filter fn body to [bytecode], default behaviour is to change stringified fn bodies to [bytecode].